src/repositories/timesheet/repository.rs

```diff
@@ -0,0 +1,125 @@
+use crate::common::DbResult;
+use crate::repositories::timesheet::models::{
+    TimesheetCreateData, TimesheetDb, TimesheetUpdateData, TimesheetWithWorkdays, Workday,
+};
+use sqlx::postgres::PgPool;
+use std::sync::Arc;
+use uuid::Uuid;
+
+#[derive(Clone)]
+pub struct TimesheetRepository {
+    pub pool: Arc<PgPool>,
+}
+
+impl TimesheetRepository {
+    pub fn new(pool: Arc<PgPool>) -> Self {
+        Self { pool }
+    }
+
+    // CRUD
+
+    pub async fn _create(&self, data: TimesheetCreateData) -> DbResult<TimesheetDb> {
+        let executor = self.pool.as_ref();
+
+        let timesheet = sqlx::query_as!(
+            TimesheetDb,
+            "INSERT INTO timesheet (start_date, end_date, total_hours, is_editable, manager_note, user_id, company_id, event_id) VALUES ($1, $2, $3, $4, $5, $6, $7, $8) RETURNING *;",
+            data.start_date,
+            data.end_date,
+            data.total_hours,
+            data.is_editable,
+            data.manager_note,
+            data.user_id,
+            data.company_id,
+            data.event_id
+        )
+        .fetch_one(executor)
+        .await?;
+
+        Ok(timesheet)
+    }
+
+    pub async fn _read_one(&self, uuid: Uuid) -> DbResult<TimesheetWithWorkdays> {
+        let executor = self.pool.as_ref();
+
+        let timesheet =
+            sqlx::query_as!(TimesheetDb, "SELECT * FROM timesheet WHERE id = $1;", uuid)
+                .fetch_one(executor)
+                .await?;
+
+        let workdays = sqlx::query_as!(
+            Workday,
+            "SELECT date, total_hours, comment, is_editable FROM workday WHERE timesheet_id = $1;",
+            uuid
+        )
+        .fetch_all(executor)
+        .await?;
+
+        let result = TimesheetWithWorkdays {
+            timesheet,
+            workdays,
+        };
+
+        Ok(result)
+    }
+
+    pub async fn _read_all(&self) -> DbResult<Vec<TimesheetDb>> {
+        let executor = self.pool.as_ref();
+
+        let timesheets = sqlx::query_as!(TimesheetDb, "SELECT * FROM timesheet;")
+            .fetch_all(executor)
+            .await?;
+
+        Ok(timesheets)
+    }
+
+    pub async fn _update(&self, uuid: Uuid, data: TimesheetUpdateData) -> DbResult<TimesheetDb> {
```

Comment:
The operation of updating a timesheet also includes updating the corresponding workdays.